packages/host/app/resources/search.ts

```diff
@@ -191,35 +163,29 @@ export class Search extends Resource<Args> {
               (
                 (
                   await Promise.allSettled(
-                    collectionDoc.data.map(async (doc) =>
-                      this.getOrCreateCardResource({ data: doc }),
+                    collectionDoc.data.map(async (jsonAPIResource) =>
+                      this.store.createSubscriber({
+                        resource: this,
+                        urlOrDoc: { data: jsonAPIResource },
+                        relativeTo: undefined,
+                        isAutoSaved: this.isAutoSaved,
+                        isLive: this.isLive,
+                      }),
                     ),
                   )
                 ).filter(
                   (p) => p.status === 'fulfilled',
-                ) as PromiseFulfilledResult<CardResource>[]
+                ) as PromiseFulfilledResult<{ card: CardDef }>[]
               ).map((p) => p.value)
             );
           }),
         ),
       );
-
-      await Promise.all(results.map((r) => r.loaded));
-      let resultsWithoutErrors = results.filter((r) => r.card && r.url);
-      let resultMap = new Map<string, CardResource>();
-      for (let resource of resultsWithoutErrors) {
-        resultMap.set(resource.url!, resource);
-      }
-      for (let url of this.currentResults.keys()) {
-        if (!resultMap.has(url)) {
-          this.currentResults.delete(url);
-        }
-      }
-      for (let [url, resource] of resultMap) {
-        if (!this.currentResults.has(url)) {
-          this.currentResults.set(url, resource);
-        }
-      }
+      this._instances.splice(
+        0,
+        this._instances.length,
+        ...results.map(({ card }) => card),
+      );
```

Comment:
so here the whole entire instance array is cleared? 

Can the card know its the same after coming back from the server? Like checking with the identity context? So, it only mutates the part that has changed. Now it looks that if I invalidate a single card, my whole list of cards which is rendered will be replaced.

Reply:
Cud it be beneficial to have a inPlaceEdit method?

Reply:
Yeah, I think I can add something that is a little more nuanced. The challenge would be to remove and insert cards in the array in the correct order. You might still have the same item in the results after the search, but now it’s the 7th result and no the 2nd result. My hope would be that glimmer can do the bookkeeping to see what has changed.  When you say that you invalidate a single card the whole list is pretended, are you actually seeing that or are you guessing that that is what is happening?

Reply:
Also, consider in the template when you actually render the results, if you use an @key param in the {{each}} it should only rerender the difference. This is how the card grid works

Reply:
Although since we are using an identity map, @key is probably superfluous since we are not creating new instance. 

2 things to keep in mind:
1. The array is being mutated not replaced
2. The items in the array are from an identity map so we are never recreating an instance that already exists. 

Given then I think that removing all the items in the array and then re-adding them back in a synchronous operation  should be stable when rendered with glimmer.

Reply:
> Although since we are using an identity map, @key is probably superfluous since we are not creating new instance.
> 
> 2 things to keep in mind:
> 
> 1. The array is being mutated not replaced
> 2. The items in the array are from an identity map so we are never recreating an instance that already exists.
> 
> Given that I think that removing all the items in the array and then re-adding them back in a synchronous operation should be stable when rendered with glimmer.

Cool. This is good to know that this pattern then works as expected